src/plots/cartesian/axis_ids.js

```diff
@@ -82,6 +91,8 @@ exports.listIds = function(gd, axLetter) {
 // optionally, id can be a subplot (ie 'x2y3') and type gets x or y from it
 exports.getFromId = function(gd, id, type) {
     var fullLayout = gd._fullLayout;
+    // remove "domain" suffix
+    id = ((id === undefined) || (typeof(id) !== 'string')) ? id : id.replace(/ *domain/, '');
```

Comment:
Since we are only interested to substitute the first acurance of ` domain`,
could we simply use `id.replace(' domain', '');` instead of the regular expression form here?